bbpower/compsep.py

```diff
@@ -742,14 +742,50 @@ def timing(self, n_eval=300):
         end = time.time()
 
         return end-start, (end-start)/n_eval
+    
+    def predicted_spectra(self):
+        """
+        Evaluates model at a single point (the MAP value read 
+        from pch.stats or the param_chains.npz) and writes 
+        predicted spectra into a numpy array with shape 
+        (nbpws, nmaps, nmaps).
+        """
+        emcee_path = '/'.join(self.get_output('param_chains').split('/')[:-1])+'/emcee.npz'
+        pch_path = self.get_output('param_chains')[:-4]+'/pch.stats'
+        par_names = self.params.p_free_names
+        map_params = np.zeros((len(par_names)))
+        # This assumes the PolyChord or emcee output parameters
+        # are the same as in self.params. TODO: check?
+        if os.path.isfile(pch_path):
+            for ip, p in enumerate(par_names):
+                with open(pch_path) as f:
+                    for line in f:
+                        if line.startswith("Dim No."):
+                            break
+                    for ipar in range(len(par_names)):
+                        l = f.readline() 
+                        if ipar == ip:
+                            map_params[ip] = float(l.split()[1])
+        elif os.path.isfile(emcee_path):
+            chain = np.load(emcee_path)['chain']
+            for ip, p in enumerate(par_names):
+                map_params[ip] = np.mean(chain[:,:,ip].flatten())
+        else:
+            raise ImportError('No PolyChord or emcee output found')
+        
+        model_cls = self.model(self.params.build_params(map_params))
+        np.save('/'.join(self.get_output('param_chains').split('/')[:-1])+'/cells_model.npy',
+                model_cls)
```

Comment:
Could we save this as npz with a bit more information? E.g. the ordering of the frequencies assumed (e.g. names of the bands?) and the effective ells of the bandpowers. This is so it's easier to plot things directly from this file.

Reply:
Yes! I'll take care of it.

Reply:
@kwolz just another ping about this one. As soon as that's done we can merge, I think.